PIMbench/cpp-vgg13/baselines/vgg13.py

```diff
@@ -0,0 +1,102 @@
+# Inferencing using a pre-trained VGG13 on CPU or GPU with CUDA support
+import numpy as np
+import torch
+import torch.nn as nn
+from torchvision import datasets, transforms, models
+from torch.utils.data import Subset
+import time
+import argparse
+
+# Command-line argument parsing
+parser = argparse.ArgumentParser(description='VGG13 Inference')
+parser.add_argument('-s', '--device', required=True, choices=['cpu', 'cuda'], help='Device to use for inference (cpu/cuda)')
+parser.add_argument('-n', '--num_images', type=int, default=1000, help='Number of images to test')
+args = parser.parse_args()
+
+deviceStr = args.device
+if deviceStr != 'cpu' and deviceStr != 'cuda':
+    print("[ERROR] Invalid device! Must be either cpu or cuda.")
+    exit(-1)
+if deviceStr == 'cuda' and not torch.cuda.is_available():
+    print("[WARNING] cuda is not available. Falling back to cpu.")
+    deviceStr = 'cpu'
+print("[INFO] Using " + deviceStr)
+device = torch.device(deviceStr)
+
+def data_loader(data_dir, batch_size, num_images, shuffle=True, test=False):
+    print("[INFO] Starting data loader")
+    normalize = transforms.Normalize(
+        mean=[0.485, 0.456, 0.406],
+        std=[0.229, 0.224, 0.225],
+    )
+
+    transform = transforms.Compose([
+            transforms.Resize((224, 224)),
+            transforms.ToTensor(),
+            normalize,
+    ])
+
+    if test:
+        dataset = datasets.CIFAR100(root=data_dir, train=False, download=True, transform=transform)
```

Comment:
Can you please mention in the readme how much storage is needed for this?

Reply:
Updated the comments in the README file. Kindly check.